code/game/gamemodes/cm_process.dm

```diff
@@ -268,8 +269,10 @@ dat += " You failed to evacuate \the [MAIN_SHIP_NAME]"
 					if(0) //nullspace
 						if(M.loc && M.loc.z == 3) //in a closet or vent
 							numXenosShip++
+							xenoLocationsS += M.loc.loc.name
```

Comment:
Make the `if(M.loc)` check run only once.
Or even better, use `get_area(M)`,which is cleaner and works a bit more reliably than checking the `locs` and accounting for `loc` inceptions.